mirrord/operator/src/crd/mysql_branching.rs

```diff
@@ -85,3 +94,77 @@ pub struct SessionInfo {
     /// Owner info of the session.
     pub owner: SessionOwner,
 }
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct BranchCopyConfig {
+    /// The default copy mode for the branch.
+    pub mode: BranchCopyMode,
+
+    /// An optional list of tables whose schema and data will be copied based on their
+    /// table level copy config. Only compatible with `Empty` and `Schema` copy mode.
+    pub tables: BTreeMap<String, TableCopyConfig>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub enum BranchCopyMode {
+    /// Create an empty database only.
+    Empty,
+    /// Create a database with all tables' schema copied from the source database.
+    Schema,
+    /// Create a database and copy all tables' schema and data fro mthe source database.
+    Data,
```

Comment:
maybe document that specifying `tables` is not compatible with this mode, and say what happens when specified (ignored? error?).

Reply:
Sounds good, it is important to highlight this is not compatible with `tables`.